mycroft/gui/extensions.py

```diff
@@ -47,29 +48,42 @@ def activate_extension(self, extension_id):
             self.extension = BigscreenExtension(self.bus, self.gui)
         elif extension_id == "mobile":
             self.extension = MobileExtension(self.bus, self.gui)
+        elif extension_id == "plasmoid":
+            self.extension = PlasmoidExtension(self.bus, self.gui)
         else:
             self.extension = GenericExtension(self.bus, self.gui)
 
         LOG.info(f"Extensions Manager: Activated Extension {extension_id}")
         self.bus.emit(
             Message("extension.manager.activated", {"id": extension_id}))
 
+        def signal_available(message=None):
+            message = message or Message("")
+            self.bus.emit(message.forward("mycroft.gui.available",
+                                          {"permanent": self.extension.permanent}))
+
+        if self.extension.preload_gui:
+            signal_available()
+        else:
+            self.bus.on("mycroft.gui.connected", signal_available)
```

Comment:
Should this ever happen more than once?

Reply:
yes, anytime we go from 0 to 1 or more guis connected, because skills unload when we go from 1+ to 0 gui connections